sdks/smart-wallet-sdk/src/smartWallet.ts

```diff
@@ -0,0 +1,77 @@
+import { Interface } from '@ethersproject/abi'
+import { ChainId } from '@uniswap/sdk-core'
+
+import { abi } from '../abis/MinimalDelegation.json'
+
+import { ModeType, SMART_WALLET_ADDRESSES } from './constants'
+import { Call, MethodParameters, ExecuteOptions } from './types'
+import { ExecuteCallPlanner, ModeEncoder } from './utils'
+
+/**
+ * Main SDK class for interacting with ERC7821-compatible smart wallets
+ */
+export class SmartWallet {
+  /**
+   * Interface for the Smart Wallet contract (will be initialized from ABI)
+   */
+  public static INTERFACE: Interface = new Interface(abi)
+
+  /**
+   * Creates method parameters for executing a batch of calls through a smart wallet
+   * @param calls Array of calls to encode
+   * @param options Execution options
+   * @param chainId The chain ID for the calls
+   * @returns Method parameters with calldata and value
+   */
+  public static encodeExecute(calls: Call[], options: ExecuteOptions = {}): MethodParameters {
```

Comment:
execute options should be optional, or default to false? I think since the main use case is basically not an advanced mode, we should just expose a function thats the easiest to ingest from the wallet side

Reply:
just giving some reasoning for why in the sdk doc it just accepted an array of calls. My thinking is just that starting with the simplest possible interface we could give to wallet seems like it makes the most since given 5792 can't even handle the advanced cases yet?

Reply:
It is optional in the current interface, my reasoning for including them even for the basic case is for the BATCHED_CALL_CAN_REVERT mode, which doesn't support partial reverts at the per call level (advancedCall which requires you to specify per call) but rather over the entire batch. Seems easy enough to add the flexibility for now